libs/ml/include/ml/core/node.hpp

```diff
@@ -397,14 +396,17 @@ void Node<TensorType>::ResetCache(bool input_size_changed)
  * @param op_ptr
  */
 template <typename TensorType>
-void Node<TensorType>::SetNodeSaveableParams(NodeSaveableParams<TensorType> const &nsp,
-                                             std::shared_ptr<ops::Ops<TensorType>> op_ptr)
+void Node<TensorType>::SetNodeSaveableParams(NodeSaveableParams<TensorType> const &       nsp,
+                                             std::shared_ptr<ops::Ops<TensorType>> const &op_ptr)
 {
   name_                 = nsp.name;
   cached_output_status_ = CachedOutputState::CHANGED_SIZE;
   operation_type_       = nsp.operation_type;
   op_ptr_               = op_ptr;
 }
 
+std::string OutputShapeAsString(fetch::math::SizeVector const &out_shape);
```

Comment:
these two functions look like utilities.
Firstly, I believe they should be static functions.
Secondly, they should either live only in the cpp (if we're sure they will never be invoked from anywhere except node), or they should move to a more general utilities location (like ml/utilities).